generate.cr

```diff
@@ -467,13 +467,39 @@ class COverload
       ret_s = to_tuple(rets) || "Void"
       any_outputter = self.parent.overloads.any?(&.input_output_arg?)
       self.comment(&block)
-      yield %(  #{"pointer_wrapper " if any_outputter}def #{"self." if !inside_class}#{self.name(ctx)}(#{args.join(", ")}) : #{ret_s})
+      def_name = self.name(ctx)
+      yield %(  #{"pointer_wrapper " if any_outputter}def #{"self." if !inside_class}#{def_name}(#{args.join(", ")}) : #{ret_s})
       call = %(    LibImGui.#{self.name(Context::Lib)}(#{call_args.join(", ")}))
       call = %(    result = #{call}) if outp.first? == "result" && outp2 != ["result"]
       yield call
       yield (assert to_tuple(outp2)) unless outp2.empty? || outp2 == ["result"]
       yield %(  end)
       yield %(  {% end %}) if as_datatype
+
+      return if inside_class
+      antonyms = {"push" => "pop", "begin" => "end"}
+      names = {"push" => "with", "begin" => "within"}
+      always_pop_names = %w[begin begin_child begin_child_frame begin_disabled begin_group begin_tooltip push_allow_keyboard_focus push_button_repeat push_clip_rect push_clip_rect_full_screen push_font push_id push_item_width push_style_color push_style_var push_texture_id push_text_wrap_pos]
+      cond_pop_names = %w[begin_combo begin_drag_drop_source begin_drag_drop_target begin_list_box begin_main_menu_bar begin_menu begin_menu_bar begin_popup begin_popup_context_item begin_popup_context_void begin_popup_context_window begin_popup_modal begin_tab_bar begin_tab_item begin_table tree_node]
```

Comment:
Do we also want to add `tree_push`? There aren't any uses of it in the demo, but I think it should be possible to use tree_push/tree_pop

Reply:
Yes I'll add that. Just not sure how to name that method 🤔
`with_tree` is what I was thinking mainly